### PR TITLE
Change all Snackbars to notify in whole codebase 

### LINKED ISSUE #8512: [UI]  update Snackbar Calls to Use New Notification Hooks
## Current State:
Throughout the codebase, the notistack snackbars are being used for notifications. However, we have recently improved the notification experience by developing our own custom hooks and components for handling notifications.

## Desired State:
We need to replace all previous snackBar calls with the new notification hooks that we have created.

## Previous Snackbar Usage:

```
handleClick() {
  // Previous snackBar call
  enqueueSnackbar(message, {
    variant: "info",
    action: <Button> </Button>
  });
}
```
## New Notification Hook Usage:

```
const { notify } = useNotification();

handleClick() {
  // New notification hook call
  notify({ message: "hello", event_type: EVENT_TYPES.INFO });
}
```
We also have created a withNotify HOC to use in class based components 

## Steps To Work On This :

- Identify instances where the old snackBar calls are being used throughout the codebase.
- Replace the old snackBar calls with the new notify hook calls, using the provided code snippet as a guide.
- Test the application thoroughly to ensure that notifications are being displayed correctly using the new notification hooks.
- Ensure that the useNotify hook is imported and used correctly in the components where notifications are triggered.

 

### Mockups

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Rajdip019**
**Notes for Reviewers**

This PR fixes #8512 

I have changed 3 files for now with different types and cases of changing `useSnackbar` to `notify`. Needs to confirm whether these fixes are rightly implemented or not. Then I will proceed to change the other files and add commits here.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by aabidsofi19** [comment]
did you use withNotify HOC for classbased components

File: ui/components/DashboardComponent.js [hunk 1]
```diff
@@ -541,16 +542,8 @@ class DashboardComponent extends React.Component {
...
-    this.props.enqueueSnackbar(`${msg}: ${error}`, {
-      variant : "error", preventDuplicate : true,
-      action : (key) => (
-        <IconButton key="close" aria-label="Close" color="inherit" onClick={() => self.props.closeSnackbar(key)}>
-          <CloseIcon style={iconMedium} />
-        </IconButton>
-      ),
-      autoHideDuration : 7000,
-    });
+    const notify = this.props.notify;
```
Comment (Rajdip019) [inline ui/components/DashboardComponent.js:545]: This is what I have done for the HOC. I have seen the file you have sent in the issue. Is there anything I need to change here?

**Comment by Rajdip019** [comment]
Also, a question is for close actions we need not pass anything right?  @aabidsofi19

**Comment by Rajdip019** [comment]
I have made some changes withNotify HOC for class-based components. You can check now and confirm I can proceed with changing other parts as well.

Also, a question is for close actions we need not pass anything right? @aabidsofi19

**Comment by aabidsofi19** [comment]
we dont need close action, we handle that in hook .  did you work on my previous comments .

File: ui/components/DashboardComponent.js [hunk 2]
```diff
@@ -1277,5 +1270,5 @@ const mapStateToProps = (state) => {
 };
 
 export default withStyles(styles, { withTheme : true })(
-  connect(mapStateToProps, mapDispatchToProps)(withRouter(withSnackbar(DashboardComponent)))
+  connect(mapStateToProps, mapDispatchToProps)(withRouter(withNotify(DashboardComponent)))
```
Comment (Rajdip019) [inline ui/components/DashboardComponent.js:1273]: Yes. This is the change you wanted right? @aabidsofi19

**Comment by aabidsofi19** [comment]
regarding the hook ( also react hooks in general ) should only be called  directly inside the component , not inside function but you can use the return value (notify in this case ) anywhere

**Comment by Rajdip019** [comment]
So for the functions, we can pass the `notify` returned by the `useNotification` hook. And use it inside a function. Right? @aabidsofi19

**Comment by aabidsofi19** [comment]
withNotify should only used in CBC's , regarding hooks yeah . Also pls take a look at react hooks documentation

**Comment by Rajdip019** [comment]
Got it now. I will try make a full PR by tonight.

**Comment by aabidsofi19** [comment]
great great , no need to furry ..

**Comment by Rajdip019** [comment]
I haven't changed the `snackbar` for the loading notification. Because as I have seen loading is not implemented in the `useNotification` hook. @aabidsofi19

File: ui/components/ConfirmationModal.js [hunk 3]
```diff
@@ -234,24 +237,15 @@ function ConfirmationMsg(props) {
   const handleKubernetesClick = (ctxID) => {
     updateProgress({ showProgress : true })
     pingKubernetes(
-      successHandlerGenerator(enqueueSnackbar, closeButtonForSnackbarAction(closeSnackbar), "Kubernetes pinged", () => updateProgress({ showProgress : false })),
```
Comment (aabidsofi19) [inline ui/components/ConfirmationModal.js:237]: have you changed the respective functions like successHandlerGenerator to accomodate the change

**Comment by aabidsofi19** [comment]
great .. :+1:

**Comment by Rajdip019** [comment]
Let me know if any changes are required in the PR, I will be happy to implement them.

File: ui/components/ConnectionWizard/helpers/common.js [hunk 4]
```diff
@@ -1,25 +1,22 @@
...
+import { EVENT_TYPES } from "../../../lib/event-types";
 
 export const closeButtonForSnackbarAction = (closeSnackbar) => (key) => (
   <IconButton key="close" aria-label="Close" color="inherit" onClick={() => closeSnackbar(key)}>
     <CloseIcon />
   </IconButton>
 )
 
-export const successHandlerGenerator = (snackbar, action, msg, cb) => (res) => {
+export const successHandlerGenerator = (notify, msg, cb) => (res) => {
```
Comment (Rajdip019) [inline ui/components/ConnectionWizard/helpers/common.js:12]: Yes here are changes. @aabidsofi19

**Comment by aabidsofi19** [comment]
@gopivaibhav  @EraKin575  You might help reviewing these changes

**Comment by EraKin575** [comment]
@aabidsofi19 is the time duration for each notification same?

**Comment by aabidsofi19** [comment]
> @aabidsofi19 is the time duration for each notification same?

yeah , as of now .

**Comment by Rajdip019** [comment]
Maybe we can add a new optional field to actually give `timeDuration`. If it's given it will be set to that. Otherwise, we can use a default time duration of `3000` maybe. I can do that in a next upgrade PR or in this PR as well if you want @aabidsofi19

**Comment by aabidsofi19** [comment]
Good suggestion , but don't need it right now .  You got a great pr to talk about on meshery dev call ..

**Comment by Rajdip019** [comment]
Is there anything I can help with while this PR is getting reviewed?? @aabidsofi19 @leecalcote

**Comment by leecalcote** [comment]
Perhaps, a bigger answer than you’re looking for, @Rajdip019, but we need to migrate components away from class-based to functional components

**Comment by Rajdip019** [comment]
> Perhaps, a bigger answer than you’re looking for, @Rajdip019, but we need to migrate components away from class-based to functional components. I encourage you to pick one and start.

I would love to do that. I can take up the work of changing all class-based components to function-based components.

So this is what I am thinking...
First of all, I will start with two easy components. 
- `CustomToolbar.js`
- `Error boundary`

I am starting to work on them and, I am raising an issue to change them away from class-based components, you can assign that to me.

After these two are done I will raise an Issue to move all other components and start working on it. Is it a good way to go with it @leecalcote ??

**Comment by aabidsofi19** [comment]
@Rajdip019  start with custom toolbar , error boundaries are specific to cbc's ..

**Comment by Rajdip019** [comment]
Solved the merge conflicts and raised PR #8565 for the custom toolbar component as well. @aabidsofi19

**Comment by vishalvivekm** [comment]
@Rajdip019 Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Pull request merged**